Dockerfile

```diff
@@ -1,10 +1,27 @@
-FROM rust:alpine AS build
+FROM --platform=$BUILDPLATFORM tonistiigi/xx AS xx
+FROM --platform=$BUILDPLATFORM rust:alpine AS build
+COPY --from=xx / /
+
+RUN apk add clang lld
 COPY . /app
 WORKDIR /app
-RUN apk add --update musl-dev openssl-dev openssl-libs-static \
-    && CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse \
-    OPENSSL_NO_PKG_CONFIG=1 OPENSSL_STATIC=1 OPENSSL_DIR=/usr/ \
-    cargo build -p typst-cli --release
+RUN --mount=type=cache,target=/root/.cargo/git/db \
+    --mount=type=cache,target=/root/.cargo/registry/cache \
+    --mount=type=cache,target=/root/.cargo/registry/index \
+    CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse \
+    cargo fetch
+
+ARG TARGETPLATFORM
+RUN echo a
+RUN xx-apk add --update musl-dev openssl-dev openssl-libs-static
+RUN --mount=type=cache,target=/root/.cargo/git/db \
+    --mount=type=cache,target=/root/.cargo/registry/cache \
+    --mount=type=cache,target=/root/.cargo/registry/index \
```

Comment:
I saw that Docker's default Rust Dockerfile caches the main target directory here, too. Is that something you'd consider?

Reply:
I guess it could speed up the build, but I'm not a big fan of caching build artifacts in CI. If you somehow get something broken cached all your subsequent build could also  be broken.

Currently this Dockerfile only caches the dependencies.